include/vast/Dialect/Unsupported/UnsupportedTypes.td

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2023-present, Trail of Bits, Inc.
+
+#ifndef VAST_DIALECT_IR_UNSUPPORTED_TYPES
+#define VAST_DIALECT_IR_UNSUPPORTED_TYPES
+
+def UnsupportedType
+    : Unsupported_Type< "Unsupported" > {
+  let mnemonic = "type";
+  let parameters = (ins "Type":$elementType);
+  let assemblyFormat = "`<` $elementType `>`";
+}
+
+def UnsupportedVoidType : Unsupported_Type< "UnsupportedVoid" > {
```

Comment:
when do you need such type?

Reply:
I think it can be merged to single `UnsupportedType` where you make `elementType` optional:

```
  let assemblyFormat = "`<` ( $elementType^ )? `>`";
```

or at least change the name as `UnsupportedVoidType` is a bit confusing.
Use something like `UnsupportedOpaqueType`

Reply:
I removed `UnsupportedVoidType` and used the `OptionalParameter` for `elementType`. We can use it for generating types in opaque format as well where we don't visit the underlying types.